Add support for zwave_js controller firmware updates

raman325 opened the pull request:
## Proposed change
Adds support for manual zwave_js controller firmware updates in the UI. Depends on the following PRs:
- https://github.com/home-assistant/core/pull/87239
- https://github.com/home-assistant/core/pull/87348

Will move this out of draft when the ready PRs are merged as it will likely require a merge conflict fix before this can be used and requires some additional logic to handle this https://github.com/home-assistant/frontend/pull/15517 as the controller node should not have this option in this UI

## Type of change

<!--
  What type of change does your PR introduce to the Home Assistant frontend?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR.
-->

```yaml

```

## Additional information

<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

<!--
  Thank you for contributing <3
-->

[docs-repository]: https://github.com/home-assistant/home-assistant.io

raman325 committed:
Add support for zwave_js controller firmware updates

bramkragten commented on src/panels/config/devices/device-detail/integration-elements/zwave_js/device-actions.ts:
```diff
@@ -43,59 +43,68 @@ export const getZwaveDeviceActions = async (
-  ];
+  const actions: DeviceAction[] = [];
+
+  if (!nodeStatus.is_controller_node) {
```
Why this extra if, if we are only adding actions when not `is_controller_node`

raman325 replied:
we are adding actions if there is a controller node just for the firmware update

raman325 replied:
is this possibly the confusing part because I moved the not operator outside of each individual check? I can move it back inside so each individual statement is easier to understand: https://github.com/home-assistant/frontend/pull/15515/files#diff-4834eb73ab93f8dfa87c03847b42ab8539d6b51909e48038a784dc80d08e2472R103-R106

raman325 committed:
Update controller message

raman325 committed:
caps

raman325 commented:
@kpine tested all three PRs E2E and everything looks functional!

raman325 committed:
Move constants out of render, update strings

raman325 commented:
FYI I created some new strings for controller specific text, moved some stuff out of render that doesnt need to be there, and rebased. The result of the changes are in @kpine 's comment

bramkragten commented on src/panels/config/integrations/integration-panels/zwave_js/dialog-zwave_js-update-firmware-node.ts:
```diff
@@ -253,6 +260,19 @@ class DialogZWaveJSUpdateFirmwareNode extends LitElement {
       fetchZwaveNodeStatus(this.hass, this.device!.id),
       fetchZwaveIsNodeFirmwareUpdateInProgress(this.hass, this.device!.id),
     ]);
+    if (this._nodeStatus.is_controller_node) {
+      this._localizationKeySuffix = "_controller";
+      this._abortFirmwareUpdateButton = html``;
+    } else {
+      this._localizationKeySuffix = "";
+      this._abortFirmwareUpdateButton = html`
+        <mwc-button slot="primaryAction" @click=${this._abortFirmwareUpdate}>
+          ${this.hass.localize(
+            "ui.panel.config.zwave_js.update_firmware.abort"
+          )}
+        </mwc-button>
+      `;
+    }
```
This could have stayed in render 🙃

raman325 replied:
lol noted, moving back

raman325 replied:
done

raman325 committed:
Move vars back to render
